File: 2022-1-25/Reverse_String.cpp
#include <vector>

using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        unsigned int first = 0, last = s.size() - 1;
        while (first < last) {
            swap(first++, last--, s);
        }
    }
    
    void swap(unsigned int c, unsigned int h, vector<char>& s) {
        char temp = s[h];
        s[h] = s[c];
        s[c] = temp;
    }
};

File: 2022-01-25/ReverseString.java
class Solution {
    public void reverseString(char[] s) {
        for(int i=0; i<s.length/2; i++){    
            char temp = s[i];
            s[i] = s[s.length-1-i];    
            s[s.length-1-i] = temp; 
        }
    }
}
